infra/src/Pos/Infra/Recovery/Info.hs

```diff
@@ -76,16 +85,61 @@ class Monad m => MonadRecoveryInfo m where
     -- are enumerated in 'SyncStatus'.
     getSyncStatus :: SlotCount -> m SyncStatus
 
+type MonadRecoveryInfoConstraints ctx m =
```

Comment:
@erikd @parsonsmatt @mhuesch Thoughts on this? The need for this came about as the result of reuniting the `MonadRecoveryInfo` instance below (line 97). These constraints still need to be satisfied for functions which previously had a constraint of `MonadRecoveryInfo` (from when it was an orphan). I wasn't too sure how else I should handle this.

Reply:
Wow, this is kind of interesting. `MonadRecoveryInfo` is a typeclass that has a single method and a single implementation. In functional programming, we call this a *function!*

I think the correct solution here is to:
* Remove `MonadRecoveryInfoConstraints` (its un-needed)
* Change `MonadRecoveryInfo` into a set of type constraints.
* Turn `getSyncStatus` into a function.

The second part might be something like:
```
type MonadRecoveryInfo m =
         ( Monad m
         , MonadIO m
         , MonadDBRead m
         , MonadSlots ctx m
         , MonadReader ctx m
         , HasLens RecoveryHeaderTag ctx RecoveryHeader
         , HasProtocolConstants
         )
```
Does that make sense?

Reply:
Also, everywhere that currently has `MonadRecoveryInfo m` in  a constraint now needs `MonadRecoveryInfo ctx m`.

Reply:
Also, somewhere I think you added a comment about `Mockable` which @mhuesch removed in the last couple of days :)